package/utils/mdadm/files/mdadm.init

```diff
@@ -1,7 +1,7 @@
 #!/bin/sh /etc/rc.common
 
-START=13
-STOP=98
+START=12
+STOP=99
```

Comment:
Perhaps umount should be moved up one ... the problem is that at 99 I'm not sure it's guaranteed to occur before the done script.

Reply:
Yeah, it was probably 6-10 years ago that I wrote up https://openwrt.org/docs/techref/requirements.boot.process, but was talking with a fellow by the name of Bernard Loos(sp?) about the need to deal with dependencies since that initial document was exposing some ordering challenges already.  Part of the issue for OpenWrt is having an init system that doesn't need an initramfs (for space/waste reasons), which is not a challenge systemd even attempts (ot my knowledge).  I'd have to dig to find those discussion / design documents, but we were thinking about this problem before systemd came around, but 'reel jobs' interfered with actually working on the problem.  Unfortunately I don't think I ever posted the dependency bits on the wiki, else procd might have supported them.

Reply:
Thanks for sharing that.  It helped me understand where you're coming from.  I completely understand the challenges involved.  I've been programming embedded systems for decades, with a sharp focus on Linux.  I have a reel job too & know how it goes.

OpenWrt 18 already has most of what [systemd requires](https://github.com/systemd/systemd/blob/master/README).  Others have done it and [do it with busybox](https://github.com/profusion/demystifying-systemd-for-embedded-systems).  The nay saying is tiresome & the writing has been all over the walls for years.  Every modern distro uses it and for good reason.  

I'm glad I learned it.  I've grow to appreciate it and my finger memory is slowly forgetting _service <name> start_; that's my biggest complaint.  So many years of _service_ and with systemd, they flipped the arguments to _systemctl start <name>_ (which, imho, is better because there are less keystrokes to backspace to stop/start a service, etc.  Yes, it brings additional complexity, with the benefit of great new capabilities.  Maybe I'll whip up a systemd branch for Openwrt in my infinite spare time. ;)

At the moment, I want to get through this PR and continue learning how you all like things done.  So far, I'm impressed.  Back to the task at hand; incremental change.  Good, better, best.  This is the result of my init discovery with a fairly broad proposal.   While systemd may be best, this is certainly better.

```
v18.06.x & master (proposal)
============================
network                        START=12 STOP=93 * can network be up this early & stay up a little later?
blockd.init                    START=41 STOP=94 * must start after fstab; stop before lvm2
kdump.init                     START=41 STOP=96 * must start after fstab; must stop before umount
umount                         START=NO STOP=96 * must stop before mdadm
lvm2.init                      START=15 STOP=95 * must start before fstab; must stop after umount, but before cryptsetup & mdadm
cryptsetup.init                START=14 STOP=96 * must start before lvm2; second; must stop after umount & lvm2, but before mdadm
**mdadm.init                     START=13 STOP=99 * must start before cryptsetup; must stop (last) after umount, cryptsetup, & lvm2**
done                           START=99         * done nice coceptual name, but should still run dead last; i.e. S99Zz

v18.06.1 & master (current)
===========================
cryptsetup.init                START=?? STOP=?? * Bug. cryptsetup.init is missing entirely
umount                         START=NO STOP=99 * Bug. other block devices must stop afterwards.
boot                           START=10 STOP=98 * OK? point of reference; runs /sbin/mount_root; has STOP value set, though unused?
sysctl                         START=11         * OK. point of reference
mdadm.init                     START=13 STOP=98 * Bug. umount is K99
lvm2.init                      START=15         * Bug. STOP undefined
network                        START=20 STOP=90 * starts after dnsmasq, dropbear, firewall, etc?  nice to have up before cryptsetup, so key files can be curl'ed?                                                                                                             
fstab.init                     START=40         * OK. point of reference; runs /sbin/block mount (and /sbin/block umount)
kdump.init                     START=41 STOP=98 * must be before umount & change if umount does
blockd.init                    START=80         * automounter.  should start much sooner, after fstab, & explicitly stop.
done                           START=95         * runs /etc/rc.local  S95?  many things are run after S95 ... START rc.local rant
led                            START=96         * rc.local rant
rssileds.init                  START=96 STOP=96 * rc.local rant
dsl_control                    START=97         * rc.local rant
linksys_recovery               START=97         * rc.local rant
hwmon_fancontrol               START=98         * rc.local rant
sysntpd                        START=98         * rc.local rant
adb-enablemodem                START=99         * rc.local rant; only this would run before S99boot
bootcount                      START=99         * rc.local rant
bugcheck.initd                 START=99         * rc.local rant
igmpproxy.init                 START=99         * rc.local rant
netdev-cpu                     START=99         * rc.local rant
omcproxy.init                  START=99         * rc.local rant
set-irq-affinity               START=99         * rc.local rant
urandom_seed                   START=99         * OK. saves /etc/urandom.seed; entropy.
```